test-integration/graql/reasoner/ReasoningIT.java

```diff
@@ -610,9 +612,32 @@ public void whenAppendingRolePlayers_queryIsRewrittenCorrectly(){
         }
     }
 
+    @Test
+    public void whenAppendingRolePlayers_DuplicatesAreCreated() {
+        for (int i = 0; i < 10; i++) {
+            System.out.println("repetition: " + i);
+            try (Session session = server.sessionWithNewKeyspace()) {
+                loadFromFileAndCommit(resourcePath, "appendingRPs.gql", session);
+                try (Transaction tx = session.readTransaction()) {
+                    Stream<ConceptMap> answers = tx.stream(Graql.parse("match " +
+                            "$r (anotherRole: $x, anotherRole: $x) isa baseRelation;get;").asGet());
```

Comment:
I think interesting things will now happen if you query for:
`$r (anotherRole: $x, anotherRole: $y) isa baseRelation;`
-> I think it will find two excess answers.
-> see rewriteToAtoms

A minimal example is :
```
duplicateRole-RolePlayer sub rule,
when {
    $r (anotherRole: $y, andYetAnotherRole: $z) isa baseRelation;
},
then {
    $r (anotherRole: $z) isa baseRelation;
};

insert

$x isa baseEntity;
$y isa baseEntity;
(anotherRole: $x, andYetAnotherRole: $y) isa baseRelation; 
```
The inferred relation has the RP configuration `(anotherRole: $x, anotherRole: $y, andYetAnotherRole: $y)`.
If you query for `$r (anotherRole: $x, anotherRole: $y) isa baseRelation;` 4 answers will be found: 
[$x/X, $y/Y], 
[$x/Y, $y/X],
[$x/X, $y/X],
[$x/Y, $y/Y],

where the last two are incorrect.

Reply:
You're right!